src/webgl/loading.js

```diff
@@ -241,6 +265,300 @@ function parseObj(model, lines) {
   return model;
 }
 
+/**
+ * STL files can be of two types, ASCII and Binary,
+ *
+ * We need to convert the arrayBuffer to an array of strings,
+ * to parse it as an ASCII file.
+ */
+function parseSTL(model, buffer) {
+  if (isBinary(buffer)) {
+    parseBinarySTL(model, buffer);
+  } else {
+    var reader = new DataView(buffer);
+    var decoder = new TextDecoder('utf-8');
```

Comment:
TextDecoder is still [flagged as experimental on MDN](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder/TextDecoder). I am not sure if this matters a ton as the only unsupported browser is IE which since we are switching to ES6 this year is less important. Edge has unknown compatibility but they are switching over to Chromium so also not super important. Just worth noting and thinking about whether there is an easy alternative. A quick google search reveals mostly polyfills which I think aren't worth it because they would just be removed later this year.

Reply:
[Edge has a compatibility in development](https://developer.microsoft.com/en-us/microsoft-edge/platform/status/encodingstandard/), and for the time being, [I will include a polyfill](https://developers.google.com/web/updates/2012/06/How-to-convert-ArrayBuffer-to-and-from-String), but the polyfill is slower.

Reply:
I don't think you have to include the polyfill. As I said, the transition to ES6 is more or less underway and I think it is okay to stop supporting IE. Maybe just check for TextDecoder and log something about an unsupported browser.